integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -683,6 +695,103 @@ async def get_all_issues(self, group: Group) -> typing.AsyncIterator[List[Issue]
             issues: List[Issue] = typing.cast(List[Issue], issues_batch)
             yield issues
 
+    def should_run_for_members(self, include_bot_members: bool, member: RESTObject):
+        return include_bot_members or not member.username.__contains__("bot")
+
+    async def enrich_object_with_members(
+        self,
+        obj: RESTObject,
+        include_inherited_members: bool = False,
+        include_bot_members: bool = True,
+        include_public_email: bool = False,
+    ) -> dict[str, Any]:
+        """
+        Enriches an object (e.g., Project or Group) with its members.
+        """
+        members_list = []
+        async for members in self.get_all_object_members(
+            obj, include_inherited_members, include_bot_members
+        ):
+            if include_public_email:
+                tasks = [
+                    self.enrich_member_with_public_email(member) for member in members
+                ]
+                members_list.extend(await asyncio.gather(*tasks))
+            else:
+                members_list.extend(member.asdict() for member in members)
+
+        obj_dict: dict[str, Any] = obj.asdict()
+        obj_dict["__members"] = members_list
```

Comment:
we are not consistent in the way that we are returning the enriched objects, in enrich_project you are returning objects while here you are returning dict, can you elaborate on why?

Reply:
In ProjectWithMembers, we call `enrich_project_with_extras` after which the results are passed down to `enrich_object_with_members`, for `enrich_object_with_members` to get an object, `enrich_project_with_extras ` needed to return an object.

Would you like me to make `enrich_object_with_members` to also manipulate the object directly and return a dict as well ?